src/test/java/seedu/address/logic/parser/AddTaskParserTest.java

```diff
@@ -0,0 +1,134 @@
+        assertIncorrect("");
+
+        // Too many arguments
+        assertIncorrect("\"a\" 4/5/2016 7:23am 6/7/2017 8:42pm extraArg");
+    }
+
+    @Test
+    public void parseAddDeadlineCommand() {
+        // All arguments provided
+        assertAddDeadlineParse("\"a\" 1/2/2000 4:30am", "a", LocalDateTime.of(2000, 2, 1, 4, 30));
+
+        // If date is not given, then it is the current date
+        assertAddDeadlineParse("\"a\" 9:56am", "a", LocalDateTime.of(2653, 9, 15, 9, 56));
+
+        // If time is not given, then it is 11:59pm
+        assertAddDeadlineParse("\"a\" 3/4/2015", "a", LocalDateTime.of(2015, 4, 3, 23, 59));
+
+        // Must have at least one of { date, time }
+        assertIncorrect("");
+
+        // Too many arguments
+        assertIncorrect("\"a\" 8/1/2012 1:49am extraArg");
+    }
+
+    @Test
+    public void parseAddFloatingTaskCommand() {
+        // All arguments provided
+        assertAddFloatingTaskParse("\"a\" p-3", "a", "3");
+
+        // If priority is not given, then it default "0"
+        assertAddFloatingTaskParse("\"a\"", "a", "0");
+
+        // Too many arguments
+        assertIncorrect("\"a\" 4 extraArg");
+    }
+
+    private void assertAddEventParse(String args, String name, LocalDateTime start, LocalDateTime end) {
+        final EventTask expected;
+        try {
+            expected = new EventTask(name, start, end);
+        } catch (IllegalValueException e) {
+            throw new AssertionError("should not happen", e);
+        }
+        final Command command = parser.parse(args);
+        assertTrue(command instanceof AddTaskCommand);
+        assertEquals(expected, ((AddTaskCommand)command).getTask());
+    }
+
+    private void assertAddDeadlineParse(String args, String name, LocalDateTime due) {
+        final DeadlineTask expected;
+        try {
+            expected = new DeadlineTask(name, due);
+        } catch (IllegalValueException e) {
+            throw new AssertionError("should not happen", e);
+        }
+
+        final Command command = parser.parse(args);
+
+        assertTrue(command instanceof AddTaskCommand);
+        assertEquals(expected, ((AddTaskCommand)command).getTask());
+    }
+
+    private void assertAddFloatingTaskParse(String args, String name, String priorityString) {
+        final FloatingTask expected;
+        final Priority priority;
+        try {
+            priority = new Priority(priorityString);
+            expected = new FloatingTask(name, priority);
+        } catch (IllegalValueException e) {
+            throw new AssertionError("should not happen", e);
+        }
+        final Command command = parser.parse(args);
+        assertTrue(command instanceof AddTaskCommand);
+        assertEquals(expected, ((AddTaskCommand)command).getTask());
+    }
+
+    private void assertIncorrect(String args) {
+        final Command command = parser.parse(args);
+        assertTrue(command instanceof IncorrectCommand);
+    }
+}
```

Comment:
Ugh, all these duplicated tests. There should be a way to share tests between this and `AddEventParserTest`, `AddDeadlineParserTest` etc.